src/ORM/Association.php

```diff
@@ -710,21 +708,19 @@ protected function _options(array $options): void
      * @param \Cake\ORM\Query $query the query to be altered to include the target table data
      * @param array $options Any extra options or overrides to be taken in account
      * @return void
-     * @throws \RuntimeException if the query builder passed does not return a query
-     * object
+     * @throws \RuntimeException Unable to build the query or associations.
      */
     public function attachTo(Query $query, array $options = []): void
     {
         $target = $this->getTarget();
-        $joinType = empty($options['joinType']) ? $this->getJoinType() : $options['joinType'];
         $table = $target->getTable();
 
         $options += [
             'includeFields' => true,
             'foreignKey' => $this->getForeignKey(),
             'conditions' => [],
+            'joinType' => $this->getJoinType(),
             'fields' => [],
-            'type' => $joinType,
             'table' => $table,
             'finder' => $this->getFinder(),
         ];
```

Comment:
This doesn't set defaults for all allowed options. I believe other helper functions check those, but will pull the defaults up to here in future changes if possible.